File: architect_api/app/schemas/project_api/project.py
from pydantic import BaseModel, ConfigDict
from app.schemas.blackboard_api.blackboard import  BlackboardTree


class Project(BaseModel):
    id: int
    name: str
    description: str
    title: str

    model_config = ConfigDict(from_attributes=True)


class ProjectTree(BaseModel):
    id: int
    name: str
    description: str
    title: str
    root_blackboard: BlackboardTree
    
    model_config = ConfigDict(from_attributes=True)
    
    

File: architect_ui/src/lib/types/project.ts
export interface Blackboard {
	id: number;
	title: string;
	content: string;
	parent_blackboard_id: number | null;
	child_blackboards: Blackboard[];
}

export interface Project {
	id: number;
	name: string;
	description: string;
	title: string;
	root_blackboard: Blackboard;
} 